odpf/stencil/v1/stencil.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+package odpf.stencil.v1;
+
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/descriptor.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+
+option go_package = "github.com/odpf/proton/stencil/v1;stencilv1";
+
+// These annotations are used when generating the OpenAPI file.
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+	info: {
+		version: "0.1.3";
+	};
+	external_docs: {
+		description: "Stencil server";
+	}
+	schemes: HTTP;
+};
+
+message Snapshot {
+	int64 id = 1;
+	string namespace = 2 [(google.api.field_behavior) = REQUIRED];
+	string name = 3;
+	string version = 4;
+	bool latest = 5;
+}
+
+message DownloadDescriptorRequest {
+	string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+	string name = 2 [(google.api.field_behavior) = REQUIRED];
+	string version = 3 [(google.api.field_behavior) = REQUIRED];
+	repeated string fullnames = 4;
+}
+
+message DownloadDescriptorResponse {
+	bytes data = 1;
+}
+
+message UploadDescriptorRequest {
+	string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+	string name = 2 [(google.api.field_behavior) = REQUIRED];
+	string version = 3 [(google.api.field_behavior) = REQUIRED];
+	bytes data = 4 [(google.api.field_behavior) = REQUIRED];
+	bool latest = 5;
+	bool dryrun = 6;
+	repeated string skiprules = 7;
```

Comment:
@harikrishnakanchi @rohilsurana Rules can go as a message in itself. Which can have sub fields `include` and `exclude`

Reply:
Adding Rules as Enum.
```
enum Rules {
	FILE_NO_BREAKING_CHANGE = 0;
	MESSAGE_NO_DELETE = 1;
	FIELD_NO_BREAKING_CHANGE = 2;
	ENUM_NO_BREAKING_CHANGE = 3;
}
```

Reply:
@harikrishnakanchi With this as well, we would still need include and exclude as part of a wrapper messages, right.

Reply:
```
...
Checks checks =7; 
}
message Checks {
  repeated string use = 1;
  repeated string exclude = 2;
}